tests/test_web/test_messaging.py

```diff
@@ -145,25 +145,25 @@ def test_get_ssl_domain_files_missing(mock_exists, cert_exists, key_exists, ca_e
     'missing_key', ('IIB_MESSAGING_CERT', 'IIB_MESSAGING_KEY', 'IIB_MESSAGING_CA')
 )
 @mock.patch('iib.web.messaging.os.path.exists')
-@mock.patch('iib.web.messaging.current_app')
-def test_get_ssl_domain_cert_config_not_set(mock_current_app, mock_exists, missing_key):
+def test_get_ssl_domain_cert_config_not_set(mock_exists, missing_key, app):
```

Comment:
What about the `app`

Reply:
`app` is a fixture defined on `tests/conftest.py` using the configuration from `iib.web.config.TestingConfig` which has the following config:

```
    DEBUG = True
    IIB_WORKER_USERNAMES = ['worker@DOMAIN.LOCAL']
    SQLALCHEMY_DATABASE_URI = f'sqlite:///{TEST_DB_FILE}'
    LOGIN_DISABLED = False
``` 

As stated in Observation 1 it's no longer possible to mock [Flask.current_app](https://stackoverflow.com/questions/57947935/mocking-current-app-in-flask-unit-pytest) so instead of doing it I'm using the fixture already made and including the configuration override which was being used in the original mock.

In this particular test case no additional configuration is required for `Flask.app` it just asserts that the returned domain is the expected one, with the `call_count == 3` and the certs/auths are correct.

But since `_get_ssl_domain` function in `iib.web.messaging` retrieves the config from `Flask.current_app.config` it's impossible to call it without a Flask's app set. Thus I'm using the already made fixture.

Reply:
Could you add a comment there so it doesn't get lost?

Reply:
Sure, I'll do it and send the changes in a moment

Reply:
@pbortlov @yashvardhannanavati I'm sending the files with comments as requested but I noticed that `safety` now complains about `celery` and `kombu` after running `tox`, no changes were made in `requirements*.txt` during this amended commit.

If I should include these dependency upgrades I'll need to retest everything again:
```
+==============================================================================+
|                                                                              |
|                               /$$$$$$            /$$                         |
|                              /$$__  $$          | $$                         |
|           /$$$$$$$  /$$$$$$ | $$  \__//$$$$$$  /$$$$$$   /$$   /$$           |
|          /$$_____/ |____  $$| $$$$   /$$__  $$|_  $$_/  | $$  | $$           |
|         |  $$$$$$   /$$$$$$$| $$_/  | $$$$$$$$  | $$    | $$  | $$           |
|          \____  $$ /$$__  $$| $$    | $$_____/  | $$ /$$| $$  | $$           |
|          /$$$$$$$/|  $$$$$$$| $$    |  $$$$$$$  |  $$$$/|  $$$$$$$           |
|         |_______/  \_______/|__/     \_______/   \___/   \____  $$           |
|                                                          /$$  | $$           |
|                                                         |  $$$$$$/           |
|  by pyup.io                                              \______/            |
|                                                                              |
+==============================================================================+
| REPORT                                                                       |
| checked 48 packages, using free DB (updated once a month)                    |
+============================+===========+==========================+==========+
| package                    | installed | affected                 | ID       |
+============================+===========+==========================+==========+
| celery                     | 4.4.0     | <5.2.0                   | 42498    |
| kombu                      | 4.6.7     | <5.2.1                   | 42497    |
+==============================================================================+
```

Please let me know if you think it's better to leave these packages since they're not part of JIRA card or if I take advantage of `pip-compile upgrade` and include this in my task and add them to this PR.

Thanks,

Reply:
It would be great to have NO safety comaplains. If you can upgrade :), THank you

Reply:
Well.... Now we have another problem.

I'm doing `pip-compile upgrade` for `celery`and `kombu` and they got the correct versions on `py38` only. 

On `py36` they get stuck at `celery==5.1.2` and `komu==5.1.0` which are both `< 5.2` and `security` complains.

According to this the changelogs of [celery](https://docs.celeryproject.org/en/stable/changelog.html) and [kombu](https://docs.celeryproject.org/projects/kombu/en/stable/changelog.html) the support for `py36` is dropped on 5.2, just `CTRL+F` for `drop python` to see.

I'll run the `build/run` tests again and send the amended commit but I believe we will need to drop `py36` support definitively to fix Travis as it goes [EOL this month](https://endoflife.date/python).

@yashvardhannanavati @lcarva what do you think?

Reply:
+1 on dropping py36 support. But we will have to do this carefully

Reply:
But is not that too many changes for one PR? 
We already have everything prepared for running Flask 2.0 in py3.6 right @JAVGan?

Should not we merge this, test it and then move to py38. I know it will prolong the failing CI, but is in it better to split this task?

Reply:
@lipoja yes, everything is prepared to run Flask 2.0 on both `py36` and `py38`, just the CI will fail while we keep the `py36` supported.